src/coreclr/src/vm/comsynchronizable.cpp

```diff
@@ -1450,6 +1450,12 @@ FCIMPL0(INT32, ThreadNative::GetCurrentProcessorNumber)
 {
     FCALL_CONTRACT;
 
+#ifndef FEATURE_PAL
+    PROCESSOR_NUMBER proc_no_cpu_group;
+    GetCurrentProcessorNumberEx(&proc_no_cpu_group);
```

Comment:
Is it OK if ThreadNative::GetCurrentProcessorNumber might return an index that is larger than total number of active processors on the system?

Reply:
Yes. What we return to the user is technically “ID correlated with last core we ran on”. We may even default to ThreadID if OS API is not functional. (PAL may return -1).

Are process groups not contiguous? (1,2,3, ...)

Reply:
To answer my question - yes, OS "packs" cores into as few process groups as possible and considers topology when assigning. 
https://docs.microsoft.com/en-us/windows/win32/procthread/processor-groups